### PR TITLE
Added OneTouch functionality. Automatically falls back to token.

### PR DESCRIPTION
**Pull request by jarodreyes**
A lot of changes here, this branch should not be merged until the new tutorial is live on twilio.com/docs

### CODE REVIEW

File: api/sessions.js [hunk 1]
```diff
@@ -46,14 +47,45 @@ exports.create = function(request, response) {
...
+        if (err || !user) return invalid();
+        user.authyStatus = request.body.status;
+        user.save();
+    });
+    response.end();
+};
+
+// Internal endpoint for checking the status of OneTouch
+exports.authyStatus = function(request, response) {
+    var status = request.user.authyStatus;
```
Comment (kwhinnery) [inline api/sessions.js]: should do a quick null check:

``` javascript
var status = (request.user) ? request.user.authyStatus : 'unverified';
```

File: index.js [hunk 2]
```diff
@@ -13,6 +14,7 @@ mongoose.connect(config.mongoUrl);
 var app = express();
 app.use(express.static(path.join(__dirname, 'public')));
 app.use(urlencoded({ extended: true }));
+app.use(bodyParser.json());
```
Comment (kwhinnery) [inline index.js]: Is this necessary for every request, or can this middleware just be added for the Authy webhook route?

File: api/sessions.js (hunk 1 above)
Comment (jarodreyes) [inline api/sessions.js]: ahh got it. There might be a null user.

File: index.js (hunk 2 above)
Comment (jarodreyes) [inline index.js]: Good call.

**Comment by kwhinnery** [comment]
if we get those two comments addressed the code looks reasonable to me

**Comment by jarodreyes** [comment]
You are a taste-maker and a scholar!

**Comment by kwhinnery** [comment]
lgtm

**Comment by kwhinnery** [comment]
:+1:

**Pull request merged**